src/Cli/dotnet/commands/dotnet-sln/add/Program.cs

```diff
@@ -43,27 +49,39 @@ internal class AddProjectToSolutionCommand : CommandBase
             {
                 _relativeRootSolutionFolders = null;
             }
+
+            var slnFile = _arguments.FirstOrDefault(path => path.EndsWith(".sln"));
+            if (slnFile != null)
+            {
+                string args;
+                if (_inRoot) args = $"--{SlnAddParser.InRootOption.Name} ";
```

Comment:
nit: avoid one line if statements to keep code style consistency in the repo (Although I think we'll probably want to remove this anyways and instead give an error that doesn't include these args)